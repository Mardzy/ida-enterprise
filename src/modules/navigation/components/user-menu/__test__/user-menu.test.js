import { fromJS } from "immutable";

import UserMenu from "../user-menu";

describe("The user menu", () => {

  it("renders 3 dots while loading", () => {
    const user = fromJS({
      username: "sandor@connecterra.io"
    });

    const wrapper = mount(<UserMenu user={user} isLoading/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the user's full name", () => {
    const user = fromJS({
      givenName: "Sandor",
      surName: "Kocsis"
    });

    const wrapper = mount(<UserMenu user={user}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});