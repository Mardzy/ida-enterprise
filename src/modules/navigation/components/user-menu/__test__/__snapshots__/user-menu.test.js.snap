// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The user menu renders 3 dots while loading 1`] = `
<UserMenu
  isLoading={true}
  onLogoutClicked={[Function]}
  user={
    Immutable.Map {
      "username": "sandor@connecterra.io",
    }
  }
>
  <MenuItem>
    <div
      className="item"
      onClick={[Function]}
    >
      <MenuHeader>
        <div
          className="header"
        >
          sandor@connecterra.io
           
          <InlineDots
            amount={3}
          >
            <styled.span
              key="0"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "gQgMD",
                      "rules": Array [
                        "
  animation: 1s ",
                        Keyframes {
                          "id": "sc-keyframes-cFFqjv",
                          "inject": [Function],
                          "name": "cFFqjv",
                          "rules": Array [
                            "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                            "@keyframes cFFqjv{50%{color:transparent;}}",
                          ],
                          "toString": [Function],
                        },
                        " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gQgMD"
                >
                  .
                </span>
              </StyledComponent>
            </styled.span>
            <styled.span
              key="1"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "gQgMD",
                      "rules": Array [
                        "
  animation: 1s ",
                        Keyframes {
                          "id": "sc-keyframes-cFFqjv",
                          "inject": [Function],
                          "name": "cFFqjv",
                          "rules": Array [
                            "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                            "@keyframes cFFqjv{50%{color:transparent;}}",
                          ],
                          "toString": [Function],
                        },
                        " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gQgMD"
                >
                  .
                </span>
              </StyledComponent>
            </styled.span>
            <styled.span
              key="2"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "gQgMD",
                      "rules": Array [
                        "
  animation: 1s ",
                        Keyframes {
                          "id": "sc-keyframes-cFFqjv",
                          "inject": [Function],
                          "name": "cFFqjv",
                          "rules": Array [
                            "@-webkit-keyframes cFFqjv{50%{color:transparent;}}",
                            "@keyframes cFFqjv{50%{color:transparent;}}",
                          ],
                          "toString": [Function],
                        },
                        " infinite;
  &:nth-child(1) {
    animation-delay: 0ms;
  }
  &:nth-child(2) {
    animation-delay: 250ms;
  }
  &:nth-child(3) {
    animation-delay: 500ms;
  }
",
                      ],
                    },
                    "displayName": "styled.span",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "span",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <span
                  className="sc-bdVaJa gQgMD"
                >
                  .
                </span>
              </StyledComponent>
            </styled.span>
          </InlineDots>
        </div>
      </MenuHeader>
      <MenuMenu>
        <div
          className="menu"
        >
          <MenuItem
            onClick={[Function]}
          >
            <a
              className="item"
              onClick={[Function]}
            >
              Log out
            </a>
          </MenuItem>
        </div>
      </MenuMenu>
    </div>
  </MenuItem>
</UserMenu>
`;

exports[`The user menu renders the user's full name 1`] = `
<UserMenu
  onLogoutClicked={[Function]}
  user={
    Immutable.Map {
      "givenName": "Sandor",
      "surName": "Kocsis",
    }
  }
>
  <MenuItem>
    <div
      className="item"
      onClick={[Function]}
    >
      <MenuHeader>
        <div
          className="header"
        >
          Sandor Kocsis
           
        </div>
      </MenuHeader>
      <MenuMenu>
        <div
          className="menu"
        >
          <MenuItem
            onClick={[Function]}
          >
            <a
              className="item"
              onClick={[Function]}
            >
              Log out
            </a>
          </MenuItem>
        </div>
      </MenuMenu>
    </div>
  </MenuItem>
</UserMenu>
`;
